### DISCUSSION TITLE
Interfaces missing in the snmp polling from Akvorado

### DISCUSSION THREAD

**Discussion by zooper** [akvorado/akvorado#1170] [Q&A] [answered]
I am running a debian box as a router.  The router has 4 physical interfaces, where one interface is directly connected to the ISP.
The second interface is part of a bridge and creates vlans for my internal network. (the other two physical interfaces are not in use)

The problem I am facing is that Akovardo only manages to figure out 3 out of 4 interfaces.

If I look at the InIfName dimension (Same with OutIfName ), there are only 3 interfaces with name and one blank.
![image](https://github.com/akvorado/akvorado/assets/3600730/f85c3a84-4cba-4657-97da-47861a5bf7c4)

Running tcpdump on the router show's that the query also only contains 3 interfaces in the SNMP query.
192.168.101.90 = Akvorado
192.168.0.1 = debian router

`17:13:26.205252 IP 192.168.101.90.50824 > 192.168.0.1.161:  GetRequest(128)  .1.3.6.1.2.1.1.5.0 .1.3.6.1.2.1.2.2.1.2.9 .1.3.6.1.2.1.31.1.1.1.18.9 .1.3.6.1.2.1.31.1.1.1.15.9 .1.3.6.1.2.1.2.2.1.2.7 .1.3.6.1.2.1.31.1.1.1.18.7 .1.3.6.1.2.1.31.1.1.1.15.7
17:13:26.205309 IP 192.168.101.90.52139 > 192.168.0.1.161:  GetRequest(78)  .1.3.6.1.2.1.1.5.0 .1.3.6.1.2.1.2.2.1.2.2 .1.3.6.1.2.1.31.1.1.1.18.2 .1.3.6.1.2.1.31.1.1.1.15.2
17:13:26.205866 IP 192.168.0.1.161 > 192.168.101.90.50824:  GetResponse(149)  .1.3.6.1.2.1.1.5.0="asterix" .1.3.6.1.2.1.2.2.1.2.9="vlan30" .1.3.6.1.2.1.31.1.1.1.18.9="" .1.3.6.1.2.1.31.1.1.1.15.9=1000 .1.3.6.1.2.1.2.2.1.2.7="br1" .1.3.6.1.2.1.31.1.1.1.18.7="" .1.3.6.1.2.1.31.1.1.1.15.7=1000
17:13:26.206140 IP 192.168.0.1.161 > 192.168.101.90.52139:  GetResponse(131)  .1.3.6.1.2.1.1.5.0="asterix" .1.3.6.1.2.1.2.2.1.2.2="Intel Corporation Ethernet Controller I225-V" .1.3.6.1.2.1.31.1.1.1.18.2="" .1.3.6.1.2.1.31.1.1.1.15.2=1000`

It becomes a bit cluttered here, but the gist is:
The snmp query only includes ifindex 2,7 and 9.
The request does not include ifindex 8 for some reason. 

I am using the exakt same hsflowd conf to export my flow to prometheus, (I only added one more collector to the config, Akvorado) and there I can see all my interfaces.
![image](https://github.com/akvorado/akvorado/assets/3600730/0a221d93-40de-4ebd-8106-9cdc2a56ea7f)
For me this indicates that there shouldn't be anything wrong with the sflow agent config.

I have 2 more debian boxes where I am experience similar issues. They don't have vlan interfaces but some wireguard interfaces and gre tunnel interfaces and those are missing as well. 

I'm a bit clueless where to look next. 
Installation is done after the quick start guide and the docker compose approach.

**Comment by vincentbernat** [comment]
Maybe you could share a pcap of the sflow packet?

**Reply by zooper** [reply, marked as answer]
I should ofc have looked at that my self first. I over looked it from my side.
Yes, ifIndex 8 is missing in the sflow capture, for some reason. But that's for me to figure out.

Sorry for the noise!
